pkg/controllers/provisioning/binpacking/packable.go

```diff
@@ -198,45 +234,44 @@ func (p *Packable) validateCapacityTypes(constraints *v1alpha5.Constraints) erro
 }
 
 func (p *Packable) validateNvidiaGpus(pods []*v1.Pod) error {
-	if p.InstanceType.NvidiaGPUs().IsZero() {
-		return nil
-	}
-	for _, pod := range pods {
-		for _, container := range pod.Spec.Containers {
-			if _, ok := container.Resources.Requests[resources.NvidiaGPU]; ok {
-				return nil
-			}
-		}
+	needsGpu := p.needsResource(pods, resources.NvidiaGPU)
```

Comment:
Can you collapse this into 

```
if p.requiresResource(pods, resources.NvidiaGPU) == p.InstanceType.NvidiaGPUs().IsZero() {
  return fmt.Errorf("%s is only used if required", resources.NvidiaGPU)
}
```

Reply:
Maybe this error message: 
```
return errors.New("pod requirements do not match Nvidia GPUs")
```
"not required" could be understood as "is allowed" which is confusing to get in an error message.

Reply:
To be clear my suggestion was to not explicitly reference anything in the string and instead use the resource type. This will allow us to abstract :)